pkg/sample-plugin.go

```diff
@@ -119,17 +134,72 @@ func (td *SampleDatasource) CheckHealth(ctx context.Context, req *backend.CheckH
 	}, nil
 }
 
-type instanceSettings struct {
-	httpClient *http.Client
-}
+// SubscribeStream is called when a client wants to connect to a stream.  This callback allows sending the first message
+func (d *SampleDatasource) SubscribeStream(_ context.Context, req *backend.SubscribeStreamRequest) (*backend.SubscribeStreamResponse, error) {
+	log.DefaultLogger.Info("SubscribeStream called", "request", req)
 
-func newDataSourceInstance(setting backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
-	return &instanceSettings{
-		httpClient: &http.Client{},
+	return &backend.SubscribeStreamResponse{
+		Status: backend.SubscribeStreamStatusOK,
+		// Enabling UseRunStream will make Grafana open a unidirectional stream
+		// to consume data from a plugin while active subscribers exist.
+		UseRunStream: true,
 	}, nil
 }
 
-func (s *instanceSettings) Dispose() {
-	// Called before creatinga a new instance to allow plugin authors
-	// to cleanup.
+// RunStream is called once for any open channel.  Results are shared with everyone subscribed to the same channel
+func (d *SampleDatasource) RunStream(ctx context.Context, req *backend.RunStreamRequest, sender backend.StreamPacketSender) error {
+	log.DefaultLogger.Info("RunStream called", "request", req)
+
+	// Create the same data frame as for query data.
+	frame := data.NewFrame("response")
+
+	// Add the time dimension.
+	frame.Fields = append(frame.Fields,
+		data.NewField("time", nil, make([]time.Time, 1)),
+	)
+	// Add values dimension.
+	frame.Fields = append(frame.Fields,
+		data.NewField("values", nil, make([]int64, 1)),
+	)
```

Comment:
These can be combined, no? Or did you want to have them separated for clarity?

Reply:
Ryan fixed this in 6c1c8243956684553a48df7f5b3d6558fdb2b451